engine/src/test/java/org/hibernate/search/engine/backend/orchestration/spi/SingletonTaskTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.backend.orchestration.spi;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import org.hibernate.search.engine.reporting.EntityIndexingFailureContext;
+import org.hibernate.search.engine.reporting.FailureContext;
+import org.hibernate.search.engine.reporting.FailureHandler;
+import org.hibernate.search.util.impl.test.annotation.TestForIssue;
+
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.fail;
+
+@TestForIssue(jiraKey = "HSEARCH-4363")
+public class SingletonTaskTest {
+
+	private static final int TEST_WORK_DURATION = 10;
+
+	private SingletonTask testSubject = new SingletonTask(
+			"TestSingletonTask",
+			new TestWorker(),
+			new TestScheduler(),
+			new TestFailureHandler()
+	);
+
+	@Test
+	public void test() throws Exception {
```

Comment:
Maybe use a more meaningful test name?

```suggestion
	public void stopWhileBlockingWorkInProgress() throws Exception {
```